test/output-app/server/authorize.js

```diff
@@ -0,0 +1,164 @@
+import _ from 'lodash';
+import log from '../server/logger';
+
+// central logger for authorization checks
+export function authlog(resolver = '', mode = '', me = {}) {
+  const makeMessage = (message) => `Authorize ${mode} "${resolver}" with user "${me.username ? me.username : '<no-user>'}" ${message}`;
+  return {
+    debug: (message) => log.debug(makeMessage(message)),
+    error: (message) => {throw new Error(makeMessage(message))},
+  };
+}
+
+// returns the record, cached if already read, checks authorization if set
+// enhancement of tmeasday'findByIds
+export function findByIds(collection, ids = [], authQuery) {
```

Comment:
Should we try to move some of this stuff into some packages? I guess the question would be whether we think the user wants to edit the code or not.